offchain-modules/src/packages/xchain/eos/utils.ts

```diff
@@ -0,0 +1,15 @@
+export class EosAssetAmount {
+  Amount: string;
+  Asset: string;
+
+  constructor(amount: string, asset: string) {
+    this.Amount = amount;
+    this.Asset = asset;
+  }
+}
+
+//parseAssetAmount parse quantity "1.0000 EOS" to "1.0000" and "EOS"
+export function parseAssetAmount(amount: string, decimal: number): EosAssetAmount {
```

Comment:
1. This function can be a static function in `EosAssetAmount` class;
2. consider using regex? It's more precise and do not need `decimal` parameter. example:

```javascript
> '1.0000 EOS'.match(/^(\d+\.\d+)\s*(\w+)$/i)
[
  '1.0000 EOS',
  '1.0000',
  'EOS',
  index: 0,
  input: '1.0000 EOS',
  groups: undefined
]
```

Reply:
fixed